### ISSUE TITLE
FR: geom_lollipop

### ISSUE THREAD

**Issue by mschilli87** [ropensci/plotly#1210]
I want to suggest a feature and I would be willing to  give it a shot myself and opening a PR, but I would like to know if there is interest first.

---

I would like to see `geom_lollipop` implemented in plotly.

It is an alternative to `geom_bar` that basically combines two layers: `geom_point` and `geom_segment`.
For `ggplot`, `geom_lollipop` is implemented in the `ggalt` package.

---

Here is a minimal example:

```r
library(ggalt)
library(plotly)

mtcars <- data.frame(car=rownames(mtcars), mtcars)

base.plot <- ggplot(mtcars, aes(car, mpg))

bar.plot <- base.plot + geom_bar(stat="identity")
bar.plot + ggtitle("static bar plot")
ggplotly(bar.plot + ggtitle("reactive bar plot"))

lollipop.plot <- base.plot + geom_lollipop()
lollipop.plot + ggtitle("static lollipop plot")
#This does not work:
#ggplotly(lollipop.plot + ggtitle("reactive lollipop plot"))

lollipop.plot <- base.plot + geom_segment(aes(xend=car), yend=0) + geom_point()
lollipop.plot + ggtitle("static lollipop plot")
# Now it works:
ggplotly(lollipop.plot + ggtitle("reactive lollipop plot"))
# But hints could be improved (also show point hints on segment hover).
```
---

Please let me know if there is any interest and maybe some hints how/where to start.

**Comment by cpsievert** [comment]
This feature would be great for a first time contributor. I've actually outlined how we convert other ggalt geoms --

https://plotly-book.cpsievert.me/translating-custom-ggplot2-geoms.html
https://github.com/hrbrmstr/ggalt/blob/master/R/geom2plotly.r

As your example suggests, you could likely implement a `GeomLollipop` method for the `to_basic` generic that outputs a list of two data frames ([similar to `GeomSmooth`](https://github.com/ropensci/plotly/blob/d7eaf38e915462db75ae09790fc1b7067e10091a/R/layers2traces.R#L206)): one with class `GeomSegment` and one with class `GeomPoint`

**Comment by mschilli87** [comment]
Thx @cpsievert  for the helpful reply.
If nobody else picks this up before easter, I'll give it a shot. Until then I'll probably be to busy to 'play around'. :wink:

**Comment by jonas-hag** [comment]
Hey @mschilli87 have you already started working on it? Otherwise I'd like to give it a try :)

**Comment by mschilli87** [comment]
@jonas-hag: No. Something else came up and I won't have time before mid-May or so. So feel free to pick this up.

**Comment by cpsievert** [comment]
Awesome!

> When using "compare data on hover", the data is shown both for the point and the segment.

You could set `hoverinfo="none"` for the segments.

> Also, when using the selection tools the unselected points are transparent, but the segments are not and they seem to be in the foreground.

Unfortunately there isn't much we can do about that at the moment (besides using crosstalk, [similar to this](https://plotly-book.cpsievert.me/linking-views-without-shiny.html#nested-selections))

> Also I was wondering if to_basic.GeomLollipop should be included into plotly or into ggalt?

I think ggalt makes more sense, but would work either way

**Comment by jonas-hag** [comment]
Sorry, somehow I didn't follow up with the needed changes. @kcbioy  feel free to use the code I've already written, you find it here: https://github.com/jonas-hag/plotly/commit/171935916a64b1de89a4ad4a5e0fca7d0705880d
